src/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,318 @@
+    event Borrow(address account, uint256 amount);
+    event Repay(address account, uint256 amount);
+    event Withdraw(address account, uint256 amount);
+    event Liquidated(address account, uint256 debtAmount, uint256 collateralSeized);
+
+    error InvalidAmount(uint256 amount);
+    error InsufficentCollateral(uint256 maxCapacity, uint256 debtAmount);
+    error ExceededBorrowedAmount(address account, uint256 amountBorrowed, uint256 amountRepay);
+    error ExceededCollateralAmonut(address account, uint256 amountCollateral, uint256 collateralWithdraw);
+    error WillUnderCollaterlize(address account, uint256 withdrawalAmount);
+    error OverCollaterilized(address account);
+    
+    constructor(
+        uint256 _interestRateBps,
+        uint256 _minCollateralizationRatio,
+        uint256 _interestRatePeriod,
+
+        address _collateralToken,
+        uint256 _collateralPrice,
+
+        address _debtToken,
+        uint256 _debtPrice,
+        address _debtCollector
+
+    ) {
+        interestRateBps = _interestRateBps;
+        minCollateralizationRatio = _minCollateralizationRatio;
+        interestRatePeriod = _interestRatePeriod;
+        
+        collateralToken = IERC20(_collateralToken);
+        collateralPrice = _collateralPrice;
+
+        debtToken = IERC20(_debtToken);
+        debtPrice = _debtPrice;
+        
+        debtCollector = _debtCollector;
+    }
+
+
+    function addOperator(address _address) external override onlyOwner {
+        _addOperator(_address);
+    }
+
+    function removeOperator(address _address) external override onlyOwner {
+        _removeOperator(_address);
+    }
+
+    function setDebtPrice(uint256 _debtPrice) external onlyOperators {
+        debtPrice = _debtPrice;
+    }
+
+    function setCollateralPrice(uint256 _collateralPrice) external onlyOperators {
+        collateralPrice = _collateralPrice;
+    }
+
+    function setDebtCollector(address _debtCollector) external onlyOperators {
+        debtCollector = _debtCollector;
+    }
+
+    function setCollateralizationRatio(uint256 _minCollateralizationRatio) external onlyOperators {
+        minCollateralizationRatio = _minCollateralizationRatio;
+    }
+
+    /**
+     * @dev Get user principal amount
+     * @return principal amount
+     */
+    function getDebtAmount(address account) public view returns (uint256) {
+        return positions[account].debtAmount;
+    }
+
+    /**
+     * @dev Get user total debt incurred (principal + interest)
+     * @return total Debt
+     */
+    function getTotalDebtAmount(address account) public view returns (uint256) {
+        uint256 totalDebt = positions[account].debtAmount;
+        uint256 periodsPerYear = 365 days / interestRatePeriod;
+        uint256 periodsElapsed = block.timestamp - positions[account].createdAt; // divided by interestRatePeriod
+        totalDebt += (((totalDebt * interestRateBps) / 10000 / periodsPerYear) * periodsElapsed) / interestRatePeriod;
```

Comment:
periodsElapsed can be renamed as `secondsElapsed` now? It doesn't represent a period

Still some truncation as a result of this (slither would warn you). I think this is equivalent to:

```
    // Add on the accrued (simple) interest up to now.
    totalDebt += totalDebt * interestRateBps * secondsElapsed / 10_000 / periodsPerYear / interestRatePeriod;
```

ie move the multiplications to the front

This will also fail if createdAt == block.timestamp (div 0). An edge case to check where it should just return the `positions[account].debtAmount;`

Reply:
ok made this continuous instead of discrete